### PR TITLE
(MAINT) Bump beaker dependency to 2.12.0

### PR DESCRIPTION
**Pull request by camlow325**
This commit bumps the beaker dependency to 2.12.0.

### CODE REVIEW

**Comment by nwolfe** [comment]
:+1:

File: Gemfile [hunk 1]
```diff
@@ -15,7 +15,7 @@ gem 'jira-ruby', :group => :development
 
 group :test do
   gem 'rspec'
-  gem 'beaker', *location_for(ENV['BEAKER_VERSION'] || '~> 2.9.0')
+  gem 'beaker', *location_for(ENV['BEAKER_VERSION'] || '~> 2.12.0')
```
Comment (joshcooper) [inline Gemfile]: FWIW, we're tracking `'~> 2.8'` on the puppet side.

Comment (camlow325) [inline Gemfile]: Hmmm, is that what's actually being used for the platform side, though?  This recent job, for example, appears to be picking up 2.12.0:

https://jenkins.puppetlabs.com/view/All%20in%20One%20Agent/view/Master/view/Puppet/job/platform_aio-puppet_intn-sys_master/75/SLAVE_LABEL=beaker,TEST_TARGET=redhat-7-x86_64/consoleFull

Comment (joshcooper) [inline Gemfile]: Since we don't specify the z component, the y is allowed to float. So we're basically allowing versions >= 2.8 and less than 3

Comment (camlow325) [inline Gemfile]: Ah, k.  Missed the absent ".0" from your comment the first time I read it.  :smile:

**Pull request merged**